src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancies/Schedule.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EasyAbp.BookingService.AssetSchedules;
+using EasyAbp.BookingService.PeriodSchemes;
+
+namespace EasyAbp.BookingService.AssetOccupancies;
+
+public class Schedule
+{
+    public DateTime StartingDateTime { get; set; }
+
+    public DateTime EndingDateTime { get; set; }
+
+    public PeriodUsable PeriodUsable { get; set; }
+
+    public TimeInAdvance TimeInAdvance { get; set; }
+
+    public bool IsTimeRangeIntersected(DateTime startingDateTime, DateTime endingDateTime)
+    {
+        return !(startingDateTime >= EndingDateTime || endingDateTime <= StartingDateTime);
+    }
+
+    public bool IsContainsTimeRange(DateTime startingDateTime, DateTime endingDateTime)
+    {
+        return StartingDateTime <= startingDateTime && endingDateTime <= EndingDateTime;
+    }
+
+    public IEnumerable<Schedule> RemoveIntersectingTimeRange(DateTime startingDateTime, DateTime endingDateTime)
+    {
+        if (startingDateTime < StartingDateTime
+            && StartingDateTime < endingDateTime
+            && endingDateTime <= EndingDateTime)
+        {
+            yield return new Schedule
+            {
+                PeriodUsable = PeriodUsable,
+                TimeInAdvance = TimeInAdvance,
+                StartingDateTime = endingDateTime,
+                EndingDateTime = EndingDateTime
+            };
+        }
+        else if (StartingDateTime <= startingDateTime
+                 && startingDateTime < EndingDateTime
+                 && EndingDateTime < endingDateTime)
+        {
+            yield return new Schedule
+            {
+                PeriodUsable = PeriodUsable,
+                TimeInAdvance = TimeInAdvance,
+                StartingDateTime = StartingDateTime,
+                EndingDateTime = startingDateTime
+            };
+        }
+        else
```

Comment:
What if input `targetStartingDateTime == StartingDateTime` and input `targetEndingDateTime == EndingDateTime`?

Reply:
It will be two empty schedule, and will be removed after.